app/src/main/java/com/android/androidfundamentalsgroup1/MainActivity.java

```diff
@@ -1,33 +1,196 @@
+    private final String ANDROID_DOCS = "https://developer.android.com/";
+
+    private Button buttonSubmit;
+    private EditText email;
+    private EditText phone;
+    private CheckBox tAndC;
+    private String loginContent = "";
 
-public class MainActivity extends AppCompatActivity {
 
     private TextView oneTextView;
     private EditText editTextUserFullName;
     private CheckBox checkBoxTermsAndConditions;
     private SeekBar seekBarCountChallenges;
     private TextView textViewPurpleContent;
     private Button buttonGetContent;
+    private WebView webViewAndroid;
+    private Spinner spinnerAndroidVersions;
+    private RecyclerView recyclerViewEmails;
+
+    private List<Email> emails;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         // seteaza layout-ul asociat activatii MainActivity
         // setContentView(R.layout.activity_main);
         // asociem views_sample_1 ca layout al activitatii MainActivity
-        setContentView(R.layout.views_sample_1);
+        // setContentView(R.layout.views_sample_1);
+        // displayViewsSample1();
+
+        // run ScrollView sample
+        // setContentView(R.layout.views_sample_scroll_view);
+
+        // run WebView sample
+        // setContentView(R.layout.views_sample_web_view);
+        // loadUrlInWebView();
+
+        // run Spinner sample
+        // setContentView(R.layout.views_sample_spinner);
+        // setupSpinnerAdapter();
+
+        // run RecyclerView sample
+//        setContentView(R.layout.views_sample_recycler_view);
+//        displayEmailsList();
+
+        /////////////HOMEWORK
+
+        //view_group_challenge_1
+        //setContentView(R.layout.view_group_challenge_1);
+        //displayInfos();
+
+        //view_group_challenge_2.1
+        //setContentView(R.layout.view_group_challenge_2_screen_1);
+
+        //view_group_challenge_2.2
+        //setContentView(R.layout.view_group_challenge_2_screen_2);
+
+        //view_group_challenge_2.3
+        setContentView(R.layout.view_group_challenge_2_screen_3);
+        setupSpinnerAdapter();
+    }
+
+    private void displayInfos() {
+
+        email = findViewById(R.id.email_edit_text);
+        phone = findViewById(R.id.phone_edit_text);
+        tAndC = findViewById(R.id.t_c_check_box);
+
+        buttonSubmit = findViewById(R.id.submit);
+        buttonSubmit.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View view) {
+
+                String emailContent = email.getText().toString();
+                String phoneContent = phone.getText().toString();
+
+                if (!emailContent.isEmpty() && Patterns.EMAIL_ADDRESS.matcher(emailContent).matches()) {
+                    email.setError(null);
+                    loginContent += emailContent + "\n";
+                    if (!phoneContent.isEmpty() && Patterns.PHONE.matcher(phoneContent).matches()) {
```

Comment:
To make this code cleaner and easier to understand you could create a dedicated method for each check